src/common/libkvs/kvs_commit.h

```diff
@@ -24,10 +24,19 @@ extern "C" {
  * operations will be illegal to compact and result in an error.  Most
  * notably, if a key has data appended to it, then is overwritten in
  * the same transaction, a compaction of appends is not possible.
+ *
+ * FLUX_KVS_SYNC will ensure all data flushed to the backing store and
+ * the root reference is checkpointed.  It effectively performs a:
+ *
+ * content.flush on rank 0
+ * checkpoint on the new root reference from the commit
+ *
+ * FLUX_KVS_SYNC only works against the primary KVS namespace.
```

Comment:
It may be good to mention that users can expect an ENOSYS error if FLUX_KVS_SYNC is used but no backing store is loaded, and that this leaves KVS content unchanged (the whole commit fails).